control-plane-operator/controllers/hostedcontrolplane/etcd/etcd-init.sh

```
@@ -0,0 +1,17 @@
+# If /var/lib/data is not empty we exit early, since this means an etcd database already exists
+mkdir -p /var/lib/data
+[ "$(ls -A /var/lib/data)" ] && echo "/var/lib/data not empty, not restoring snapshot" && exit 0
+
+RESTORE_URL_VAR="RESTORE_URL_ETCD"
+RESTORE_URL=${!RESTORE_URL_VAR}
+
+# When downloading from S3, curl can succeed even if the object doesn't exist
+# and also when a pre-signed URL is expired.
+# In this case we get an XML file which can be detected with `file` so we show
+# the contents via the logs then exit with an error status
+curl -o /tmp/snapshot ${RESTORE_URL}
```

Shellcheck (0.11.0)

[info] Double quote to prevent globbing and word splitting.

(SC2086)